opm/simulators/flow/GenericOutputBlackoilModule.cpp

```diff
@@ -599,22 +617,22 @@ assignToSolution(data::Solution& sol)
     if (! this->temperature_.empty())
     {
         sol.insert("TEMP", UnitSystem::measure::temperature,
-                   std::move(this->temperature_), data::TargetType::RESTART_SOLUTION);
+                   std::move(getArray(this->temperature_)), data::TargetType::RESTART_SOLUTION);
```

Comment:
Hm, `move(getArray(...))` looks wrong to me.  The output module (regrettably) depends on `x.empty()` being `true` once the `x` array has been inserted into the `Solution` container.  I think we may have to rewrite `getArray()` as
```C++
auto getArray(std::vector<Scalar>&& input)
{
    if constexpr (std::is_same_v<Scalar, double>) {
        return input;
    }
    else {
        return std::vector<double>{input.begin(), input.end()};
    }
}
```
and then write
```C++
sol.insert(..., getArray(std::move(this->...)), ...)
```
instead.

Reply:
changed (a bit differently as discussed elsewhere).